cfme/tests/automate/custom_button/test_service_vm_custom_button.py

```diff
@@ -123,4 +174,61 @@ def test_custom_button_with_dynamic_dialog_vm():
     Bugzilla:
         1687061
     """
-    pass
+    dialog, ele_name = setup_dynamic_dialog
+    # Create button group
+    collection = appliance.collections.button_groups
+    button_gp = collection.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        type=getattr(collection, "VM_INSTANCE"),
+    )
+    request.addfinalizer(button_gp.delete_if_exists)
+
+    # Create custom button under group
+    button = button_gp.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        dialog=dialog.label,
+        system="Request",
+        request="InspectMe",
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    # Check service vm on UI and SSUI
+    service, _ = service_vm
+
+    for context in [ViaUI, ViaSSUI]:
+        with appliance.context.use(context):
+            nav_to = ssui_nav if context is ViaSSUI else ui_nav
+
+            # Navigate to VM Details page of service
+            view = nav_to(service, "VMDetails")
+
+            # Select button from custom button group dropdown
+            custom_button_group = Dropdown(view, button_gp.text)
+            assert custom_button_group.is_displayed
+            custom_button_group.item_select(button.text)
+
+            # Check default selected vm must destination vm
+            view = view.browser.create_view(DropdownDialogView)
+            serv = view.service_name(ele_name)
+            serv.dropdown.wait_displayed()
+            assert serv.dropdown.selected_option == service.vm_name
+
+            # execute button
+            assert appliance.ssh_client.run_command(
+                'echo -n "" > /var/www/miq/vmdb/log/automation.log'
+            )
+            submit = "submit" if context is ViaUI else "submit_request"
+            getattr(view, submit).click()
+
+            try:
+                wait_for(
+                    log_request_check,
```

Comment:
Reviewer: Please avoid this for now... I would move to all custom button test to `LogValidator` in single PR.

Reply:
good.

Reply:
As mentioned in BZ can we check in logs " $evm.root['vm'] " for dynamic dialog and no runtime error.Just additional checks if it can be done.

Reply:
@sshveta I would like to avoid continues automation log check for run time error even not a good idea for UI testing. I added `vm name` check after dialog invoke. It's more than sufficient for a check.